File: myonlinefm/src/main/java/com/sec/myonlinefm/updataUIListener/ObserverUIListener.java
package com.sec.myonlinefm.updataUIListener;

import com.sec.myonlinefm.data.SearchType;
import com.sec.myonlinefm.data.Station;
import com.sec.myonlinefm.data.StationProgram;

import java.util.List;
import java.util.Map;

/**
 * Created by SRC-TJ-MM-BinYang on 2018/1/31.
 */

public interface ObserverUIListener {
    void observerLiveRadioLocalUIUpData(List<Station> mStations, Map<Integer, List<StationProgram>> map);
    void observerLiveRadioCenterUIUpData(List<Station> mStations, Map<Integer, List<StationProgram>> map);
    void observerDifferentInfoUIUpData(List<Station> mStations, Map<Integer, List<StationProgram>> map);

    void observerChannelLiveUpData(List<SearchType.ChannelLive> mSearchChannelLiveList);
    void observerProgramLiveUpData(List<SearchType.ProgramLive> mSearchProgramLiveList);
    void observerOneDayProgramUpData(List<StationProgram> mOneDayPrograms);
    void observerConnectStatus(boolean isConnectNet);
}

File: myonlinefm/src/main/java/com/sec/myonlinefm/updataUIListener/ObserverUIListener.kt
package com.sec.myonlinefm.updataUIListener

import com.sec.myonlinefm.data.SearchType
import com.sec.myonlinefm.data.Station
import com.sec.myonlinefm.data.StationProgram


/**
 * Created by SRC-TJ-MM-BinYang on 2018/1/31.
 */
interface ObserverUIListener {
    fun observerLiveRadioLocalUIUpData(mStations: MutableList<Station?>?, map: MutableMap<Int?, MutableList<StationProgram?>?>?)
    fun observerLiveRadioCenterUIUpData(mStations: MutableList<Station?>?, map: MutableMap<Int?, MutableList<StationProgram?>?>?)
    fun observerDifferentInfoUIUpData(mStations: MutableList<Station?>?, map: MutableMap<Int?, MutableList<StationProgram?>?>?)
    fun observerChannelLiveUpData(mSearchChannelLiveList: MutableList<SearchType.ChannelLive?>?)
    fun observerProgramLiveUpData(mSearchProgramLiveList: MutableList<SearchType.ProgramLive?>?)
    fun observerOneDayProgramUpData(mOneDayPrograms: MutableList<StationProgram?>?)
    fun observerConnectStatus(isConnectNet: Boolean)
}